packages/deployment/lib/configValidators.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const get = require('lodash.get');
+/**
+ * @param  {Object} config kes configuration object
+ *
+ */
+function validateWorkflowDefinedLambdas(config) {
+  const lambdaResourceMatch = /\$\{(.*)LambdaFunction\.Arn\}/;
+  const stepFunctions = get(config, 'stepFunctions', {});
+  const stepFunctionStates = Object.values(stepFunctions).map((sf) => Object.values(sf.States));
+  const lambdas = Object.keys(config.lambdas);
+
+  let resources = [].concat(...stepFunctionStates).reduce((result, cfg) => {
+    if (cfg.Type === 'Task') {
+      const lambdaArnMatch = cfg.Resource.match(lambdaResourceMatch);
+      if (lambdaArnMatch) {
+        result.push(lambdaArnMatch[1]);
+      }
+    }
+    return result;
+  }, []);
+  resources = [...new Set(resources)];
```

Comment:
Oooh, `Set`. I haven't seen it used in the wild before. The purpose is to make `resources` unique, right? If so, could the same thing not be accomplished by checking if the `lambdaArnMatch` is already in `result` on line 17 before adding it?

Reply:
and in fact, we could throw the error there while we're at it.

Reply:
Will the throw "bubble" out of the anonymous function scope for `reduce`?

Reply:
No, but we can return a structure that can evaluate to do or do not throw error.     That's likely less clean though, if more performant.    Performance not being a significant issue here, that's probably not worth it.

Reply:
In either case, good point : 96b53c5